File: javascript/stack.js
'use strict'

const LinkedList = require ('./linked_list')

class Stack {
    constructor(iterable=null){
        this.func = new linkedlist.LinkedList()
        if(Array.isArray(iterable)){
            iterable.map(x => this.push(x));
        }
    }

push(val){
    return this.func.push(val)
}

pop(){
    return this.func.pop()
}

size(){
    return this.func.size()
}

}

module.exports = Stack

File: src/stack.py
"""Stack data structure."""
from linked_list import LinkedList


class Stack(object):
    """Stack object class."""

    def __init__(self, iterable=None):
        """Init new stack instance."""
        self._stck = LinkedList(iterable)

    def push(self, val):
        """Push new item on stack."""
        self._stck.push(val)

    def pop(self):
        """Pop last-in item on stack."""
        try:
            return self._stck.pop()
        except IndexError:
            raise IndexError('Cannot pop from an empty stack.')

    def __len__(self):
        """Return size of stack with len() function."""
        return len(self._stck)
